### PR TITLE
CATO-3761 new boxes, stuff added to directors

### PR DESCRIPTION
**Pull request by pncampbell**

### CODE REVIEW

File: src/main/scala/uk/gov/hmrc/ct/accounts/frs10x/ACQ8003.scala [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+import uk.gov.hmrc.ct.box.retriever.FilingAttributesBoxValueRetriever
+import uk.gov.hmrc.ct.box._
+
+case class ACQ8003(value: Option[Boolean]) extends CtBoxIdentifier(name = "Did any of these people become a director in this period?")
+  with CtOptionalBoolean
+  with Input
+  with ValidatableBox[Frs10xAccountsBoxRetriever with FilingAttributesBoxValueRetriever]
+  with DirectorsReportEnabled {
+  override def validate(boxRetriever: Frs10xAccountsBoxRetriever with FilingAttributesBoxValueRetriever): Set[CtValidation] =
+    if (directorsReportEnabled(boxRetriever))
```
Comment (liquidarmour) [inline src/main/scala/uk/gov/hmrc/ct/accounts/frs10x/ACQ8003.scala]: failIf ?

File: src/main/scala/uk/gov/hmrc/ct/accounts/frs10x/ACQ8009.scala [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+import uk.gov.hmrc.ct.box._
+import uk.gov.hmrc.ct.box.retriever.FilingAttributesBoxValueRetriever
+
+case class ACQ8009(value: Option[Boolean]) extends CtBoxIdentifier(name = "Did any of these people resign in this period?")
+  with CtOptionalBoolean
+  with Input
+  with ValidatableBox[Frs10xAccountsBoxRetriever with FilingAttributesBoxValueRetriever]
+  with DirectorsReportEnabled {
+  override def validate(boxRetriever: Frs10xAccountsBoxRetriever with FilingAttributesBoxValueRetriever): Set[CtValidation] =
+    if (directorsReportEnabled(boxRetriever))
```
Comment (liquidarmour) [inline src/main/scala/uk/gov/hmrc/ct/accounts/frs10x/ACQ8009.scala]: fail if ?

File: src/main/scala/uk/gov/hmrc/ct/accounts/frs10x/Directors.scala [hunk 3]
```diff
@@ -32,19 +38,6 @@ case class Directors(directors: List[Director] = List.empty) extends CtBoxIdenti
     directors.foldRight(Set[CtValidation]())((dd, tail) => dd.validate(boxRetriever) ++ tail)
   }
 
-  def directorsReportEnabled(boxRetriever: Frs10xAccountsBoxRetriever with FilingAttributesBoxValueRetriever): Boolean = {
```
Comment (liquidarmour) [inline src/main/scala/uk/gov/hmrc/ct/accounts/frs10x/Directors.scala]: Do you need the

File: src/test/scala/uk/gov/hmrc/ct/accounts/frs10x/ACQ8009Spec.scala [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+      val secretary = ACQ8009(Some(true))
+
+      secretary.validate(mockBoxRetriever) shouldBe empty
+    }
+
+    "validate as mandatory" in {
+
+      val secretary = ACQ8009(None)
+
+      secretary.validate(mockBoxRetriever) shouldBe Set(CtValidation(Some("ACQ8009"),"error.ACQ8009.required",None))
```
Comment (liquidarmour) [inline src/test/scala/uk/gov/hmrc/ct/accounts/frs10x/ACQ8009Spec.scala]: spaces

File: src/test/scala/uk/gov/hmrc/ct/accounts/frs10x/ACQ8003Spec.scala [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+      val secretary = ACQ8003(Some(true))
+
+      secretary.validate(mockBoxRetriever) shouldBe empty
+    }
+
+    "validate as mandatory" in {
+
+      val secretary = ACQ8003(None)
+
+      secretary.validate(mockBoxRetriever) shouldBe Set(CtValidation(Some("ACQ8003"),"error.ACQ8003.required",None))
```
Comment (liquidarmour) [inline src/test/scala/uk/gov/hmrc/ct/accounts/frs10x/ACQ8003Spec.scala]: spaces

File: src/main/scala/uk/gov/hmrc/ct/accounts/frs10x/ACQ8003.scala (hunk 1 above)
Comment (mario-paniccia) [inline src/main/scala/uk/gov/hmrc/ct/accounts/frs10x/ACQ8003.scala]: 👍

File: src/main/scala/uk/gov/hmrc/ct/accounts/frs10x/ACQ8009.scala (hunk 2 above)
Comment (mario-paniccia) [inline src/main/scala/uk/gov/hmrc/ct/accounts/frs10x/ACQ8009.scala]: 👍

**Pull request merged**